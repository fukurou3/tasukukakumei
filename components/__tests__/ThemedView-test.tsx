import * as React from 'react';
import renderer from 'react-test-renderer';
import { useColorScheme } from 'react-native';

import { ThemedView } from '../ThemedView';

jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');
  return { ...RN, useColorScheme: jest.fn() };
});

describe('ThemedView', () => {
  it('applies light color', () => {
    (useColorScheme as jest.Mock).mockReturnValue('light');
    const tree = renderer
      .create(<ThemedView lightColor="light" darkColor="dark" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies dark color', () => {
    (useColorScheme as jest.Mock).mockReturnValue('dark');
    const tree = renderer
      .create(<ThemedView lightColor="light" darkColor="dark" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
