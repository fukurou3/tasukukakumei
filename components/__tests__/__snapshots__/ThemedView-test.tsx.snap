// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applies light color 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "light",
      },
      undefined,
    ]
  }
/>
`;

exports[`applies dark color 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "dark",
      },
      undefined,
    ]
  }
/>
`;
